File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
    int count=0;
public:
    int goodNodes(TreeNode* root) {
        countGoodNodes(root,root->val);
        return count;
    }

    void countGoodNodes(TreeNode* root,int maxm){
        if(root==nullptr)
        return;
        
        if(root->val>=maxm)
        count++;

        countGoodNodes(root->left,max(maxm,root->val));
        countGoodNodes(root->right,max(maxm,root->val));

    }
};

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int[] arr = new int[1000];
    int i=0;
    public int kthSmallest(TreeNode root, int k) {
        sortBST(root);
        return arr[k-1];
    }

    public void sortBST(TreeNode root){
        if(root==null)
        return;

        sortBST(root.left);

        arr[i++]=root.val;

        sortBST(root.right);
    }
}
